sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/SparkParserErrorStrategy.scala

```diff
@@ -49,6 +48,10 @@ class SparkRecognitionException(
       },
       Some(errorClass),
       messageParameters)
+
+  /** Construct with pure errorClass and messageParameter information.  */
+  def this(errorClass: String, messageParameters: Array[String]) =
```

Comment:
Does `SparkRecognitionException` still need to take error class name as a parameter if it's always `PARSE_SYNTAX_ERROR`?

Reply:
That's a good question. I leave it here in case that in the future we want to make more create syntax error types, or we want to handle other rest exceptions (there should still be at least one) different from the current `PARSE_SYNTAX_ERROR`.
I think I can still keep this parameter, but give this parameter a default value, so all existing calls don't need to pass the `PARSE_SYNTAX_ERROR` explicitly. Thanks.